pkg/esp32_sdk/patches/0022-spi_flash-additional-required-functions.patch

```diff
@@ -0,0 +1,63 @@
+From 3d1c57b24873657df2d9d8603f33d4cdc4e4ab6c Mon Sep 17 00:00:00 2001
+From: Gunar Schorcht <gunar@schorcht.net>
+Date: Thu, 19 May 2022 22:55:36 +0200
+Subject: [PATCH 22/23] spi_flash: additional required functions
+
+Add an alternative implementation of `spi_flash_disable_interrupts_caches_and_other_cpu` and `spi_flash_enable_interrupts_caches_and_other_cpu` if `component/spi_flash/cache_utils.c` is compiled for RIOT
+---
+ components/spi_flash/cache_utils.c | 20 ++++++++++++++++++--
+ 1 file changed, 18 insertions(+), 2 deletions(-)
+
+diff --git a/components/spi_flash/cache_utils.c b/components/spi_flash/cache_utils.c
+index 79b6f0a6854..11a6b857c89 100644
+--- a/components/spi_flash/cache_utils.c
++++ b/components/spi_flash/cache_utils.c
+@@ -62,12 +62,13 @@ static __attribute__((unused)) const char *TAG = "cache";
+ #define DPORT_CACHE_GET_VAL(cpuid) (cpuid == 0) ? DPORT_CACHE_VAL(PRO) : DPORT_CACHE_VAL(APP)
+ #define DPORT_CACHE_GET_MASK(cpuid) (cpuid == 0) ? DPORT_CACHE_MASK(PRO) : DPORT_CACHE_MASK(APP)
+ 
+-#ifndef RIOT_VERSION
```

Comment:
Why not fix `pkg/esp32_sdk/patches/0010-spi_flash-disable-functions-not-required-or-not-supp.patch` instead?

Reply:
Good catch :+1: Hm, to be honest, I didn't have this option in mind. Does it matter? I can change it, but that requires at least one additional Murdock compilation, which is a bit of a gamble right now :wink:

Reply:
Hm, merging this patch with `pkg/esp32_sdk/patches/0010-spi_flash-disable-functions-not-required-or-not-supp.patch` would affect the serial number of subsequent patches which are either already merged (e.g. [#17601](https://github.com/RIOT-OS/RIOT/pull/17601)) or part of other PRs (e.g. PR #17442).

Reply:
Ok, I have changed patch `pkg/esp32_sdk/patches/0010-spi_flash-disable-functions-not-required-or-not-supp.patch` accordingly. Furthermore, I renamed the bootloader patch `pkg/esp32_sdk/patches/0024-bootloader-remove-compile-from-banner.patch` to fix the serial number.

I squashed directly to avoid an additional CI compilation.

Reply:
Changing the title of `0010-spi_flash-disable-functions-not-required-or-not-supp.patch` to something `0010-spi_flash-disable-and-replace-functions.patch` didn't work. The result was that the additional functions were lost and the compilation failed. I'm not really happy with it, because the patch now does more than just disable some features. If it makes more sense to have a separate patch for the additional functions, I can change it back.

Reply:
Are the patches OK now or should I revert the last changes?